x/uniswap/handler.go

```diff
@@ -0,0 +1,158 @@
+package uniswap
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/supply/types"
+)
+
+// NewHandler routes the messages to the handlers
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		switch msg := msg.(type) {
+		case MsgSwapOrder:
+			return HandleMsgSwapOrder(ctx, msg, k)
+		case MsgAddLiquidity:
+			return HandleMsgAddLiquidity(ctx, msg, k)
+		case MsgRemoveLiquidity:
+			return HandleMsgRemoveLiquidity(ctx, msg, k)
+		default:
+			errMsg := fmt.Sprintf("unrecognized uniswap message type: %T", msg)
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+// HandleMsgSwapOrder handler for MsgSwapOrder
+func HandleMsgSwapOrder(ctx sdk.Context, msg MsgSwapOrder, k Keeper) sdk.Result {
+	if msg.IsBuyOrder {
+		inputAmount := k.GetInputAmount(ctx, msg.SwapDenom, msg.Amount)
```

Comment:
Where does the bound check occur?? Shouldn't it be added after this line?

Reply:
Also a deadline check is needed right?

Reply:
yea, also need to check that sender has enough of coin being sold. Also need checks balance checks for adding/removing liquidity